### DISCUSSION TITLE
Elfin EW11A RS485 modbus to WiFi

### DISCUSSION THREAD

**Discussion by alanmcgore** [nathanmarlor/foxess_modbus#207] [General]
Hello, 

Just a quick post to confirm I've managed to get this working using the Elfin EW11A RS485 Modbus to WiFi connector. This is available online under a myriad of brands for a little under 30 and pretty neat, the one I bought was called a 'Hyuduo Serial Device Server Rs485 to WiFi serial server' and was easy to setup. Only 4 wires, two for power which 5v 1A will do for, so I cut up an old USB cable and plugged it into the Inverter socket, other two wires for the A and B wires to the inverter com port. 

It has a simple web interface that you can access at whatever IP address you assign it and I then configured it similarly to the TCP 410s. Setup in the HA integration as 'Other'.

Seems to be working well now, will report if there's any issues.

**Comment by canton7** [comment]
Ooh awesome, thanks!

Can you try increasing the "max read" value under "CONFIGURE"? You'll need to keep an eye on the log for any warnings / errors which result. 

This is how many registers we try and read at once. The value we use for "Other" is pretty conservative, but most adapters can handle something higher - so if we can figure out what this adapter can handle, I'll add it as an option in the setup wizard, which uses the higher value.

**Reply by canton7** [reply]
Would you mind filling in some of https://github.com/nathanmarlor/foxess_modbus/wiki/Elfin-EW11 with some screenshots of the config, etc?

**Reply by alanmcgore** [reply]
> Ooh awesome, thanks!
> 
> Can you try increasing the "max read" value under "CONFIGURE"? You'll need to keep an eye on the log for any warnings / errors which result.
> 
> This is how many registers we try and read at once. The value we use for "Other" is pretty conservative, but most adapters can handle something higher - so if we can figure out what this adapter can handle, I'll add it as an option in the setup wizard, which uses the higher value.

No problem, I've increased the Max Read value to 100 in common with some of the other adapters and everything seems fine, I've had no issues reported in the main logs and with debugging enabled for the integration I can see no errors/plenty of successful reads for the past few hours.

**Reply by alanmcgore** [reply]
> Would you mind filling in some of https://github.com/nathanmarlor/foxess_modbus/wiki/Elfin-EW11 with some screenshots of the config, etc?

I've added some more information on the Wiki page regarding wiring and setup, if I get a chance I'll add screenshots of the config as well, in the mean time I've included some rough steps for it at least.

**Reply by canton7** [reply]
Weird, it's not showing any edits here. Cheers!

Does it support UDP, do you know?

**Reply by alanmcgore** [reply]
Apologies it didn't post for whatever reason, I've updated it now!

Yes it does seem to support UDP alongside a few other options, I've not tried these as it seems to be working well as is. Though my Home Assistant has slowed somewhat with all the additional data!

<img width="1106" alt="image" src="https://user-images.githubusercontent.com/114230187/236061855-b0428ae0-e32c-427f-b65f-75bc5d0ed74c.png">

**Reply by canton7** [reply]
Fab, thanks! I'll give the option to connect over UDP as well

**Reply by canton7** [reply]
That wiki page is looking great, thanks for your work there. It's good to have a decent wifi alternative to the W610

**Comment by tomfielder** [comment]
Hi @alanmcgore. I'm having trouble with the Elfin device. I bought the same version you did, have cut the end off an old USB and wired to the power. When I plug it in, the green light comes on and I can see the wireless access point... but for some reason I can't connect. I've tried connecting from three different devices, but no joy. I've tired pressing the reload button to completely reset it, and try to get it into smart link mode... but nothing seems to work.

Do you have any tips for getting it working?

I've not actually wired it to the inverter yet, could that be why?

The only thing I can think of that's different is that I used an old two pin USB cable. Would that cause this problem though?

Thanks in advance for any tips. Cheers.

**Reply by dreampi** [reply]
This sounds like issue with your Elfin device. Try and get a replacement if possible

**Reply by alanmcgore** [reply]
Sorry for the lack of update on this. Since setup I've found the Elfin to be very reliable, the only issues I've had are around the modbus wires coming disconnected/loose. When this happens the device stops responding and loses connectivity despite showing a green light and looking like it's working. 

So if you're having any problems around connectivity check that the modbus connection with the inverter (two wires that aren't power related) is intact.

**Reply by alanmcgore** [reply]
The other 'gotcha' to consider is I believe this is only a 2.4ghz device, so you will need a 2.4ghz WiFi network (b/g/n)  for it to connect to.

**Reply by hooktheboss** [reply]
yes it is 2.4ghz only

**Comment by hnkvhf** [comment]
The HF website states that a USB charger is not sufficient.

**Comment by hooktheboss** [comment]
I bought an Elfin EW11 labelled "AYNEFY WiFi Serial Server, RS485 to WiFi Serial Server" for €22. The housing suggests that it is an original from Elfin. It has been connected to a FoxESS H3-6.0-E inverter for a week, no problems so far, seems to be very reliable.
Keep in mind you need a power supply LESS or EQUAL to 1A, Was not so easy to find one in my house, so you might order one extra. I cuted an USB charging only cable which has of course only two wires, which make it easy to wire.

**Reply by canton7** [reply]
Your power supply's current rating should be greater than or equal to the requirements on the Elfin W11 (or any other device).

A power supply which cannot supply enough current is a problem. A power supply which can supply more current than required is fine.

**Reply by hooktheboss** [reply]
Ok, maybe not less than 1A but I tried several power supplies with more than 1A, who work with e.g. smartphones. And they were not working (not even a light on on the Elfin). That is my experience.

**Reply by canton7** [reply]
Something else was wrong, then. 

USB2.0 power supplies only have a default current of 0.5A. The device has to talk to the charger to negotiate and ask it to supply more than that. If you're just taking the power pins, and you don't have a smart device which is capable of asking the power supply for more power, then the power supply was probably refusing to give the Elfin more than 0.5A, and the Elfin was failing to start.

**Reply by hooktheboss** [reply]
It is physically not possible to connect more than the two power cables of an USB 2.0 cable to the Elfin (see the connector layout of the Elfin). Let it be as it is. My power supply says on the housing, that it is 1A and is working. It is an old apple power supply.

**Reply by canton7** [reply]
A 1A power supply will be fine, if the Elfin requires 1A.

I'm just clarifying to other people who might be reading this that you need a power supply which can provide 1A or more, and not 1A or less as you incorrectly stated.

**Comment by hnkvhf** [comment]
My elfin uses 700mA so a 1A charger will work.
I use a cheap DIN15w12 with output is 12V and 1,25A  (POS POWER, Ithink)
